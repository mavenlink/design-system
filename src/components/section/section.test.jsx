import React from 'react';
import { render, screen } from '@testing-library/react';
import Section from './section.jsx';

describe('Section', () => {
  it('renders a section component', () => {
    render(<Section />);
    expect(document.body).toMatchSnapshot();
  });

  it('can have a title', () => {
    render(<Section title="Test Title" />);
    expect(screen.getByText('Test Title').tagName).toEqual('H2');
  });

  it('can have a description', () => {
    render(<Section description="This is a test description." />);
    expect(screen.getByText('This is a test description.').tagName).toEqual('P');
  });

  it('can have child elements', () => {
    render(<Section><span>Test Label</span></Section>);
    expect(screen.getByText('Test Label').tagName).toEqual('SPAN');
  });
});
