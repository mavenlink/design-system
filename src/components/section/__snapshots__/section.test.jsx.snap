// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section renders a section component 1`] = `
<body>
  <div>
    <section
      class="section"
    />
  </div>
</body>
`;
